presto-main/src/test/java/com/facebook/presto/sql/planner/assertions/PlanMatchingVisitor.java

```diff
@@ -92,7 +93,7 @@ public MatchResult visitGroupReference(GroupReference node, PlanMatchPattern pat
         if (match.isMatch()) {
             return match;
         }
-        return visitPlan(node, pattern);
+        return matchLeaf(node, pattern, pattern.shapeMatches(node));
```

Comment:
what is leaf in this context?

Reply:
This means: if `resolve(group)` did not match the expected pattern, let's try to match `group` itself. The un-resolved group can match only as a leaf.